### PR TITLE
[#2362] improvement: replace `if..else..` with Guava's `Preconditions`

### PR DESCRIPTION
**Pull request by SteNicholas**
### What changes were proposed in this pull request?

Replace `if..else..` with Guava's `Preconditions`.

Follow up #2375.

### Why are the changes needed?

We'd better using Guava's `Preconditions`, rather than `if..else...`, which is not elegant. Check all `if...else..` block with `throw new IllegalArgumentException` patterns to find the blocker which is replaced with Guava's `Preconditions`:

Fix: #2362

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

`./gradlew build`

### CODE REVIEW

**Comment by SteNicholas** [comment]
cc @yuqi1129.

**Comment by SteNicholas** [comment]
cc @yuqi1129, @jerryshao.

File: catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveCatalogOperations.java [hunk 1]
```diff
@@ -883,11 +878,11 @@ private int columnPosition(List<FieldSchema> columns, TableChange.ColumnPosition
       int indexOfColumn = indexOfColumn(columns, afterColumn);
       Preconditions.checkArgument(indexOfColumn != -1, "Column does not exist: " + afterColumn);
       return indexOfColumn + 1;
-    } else if (position instanceof TableChange.First) {
-      return 0;
     } else {
-      throw new UnsupportedOperationException(
+      Preconditions.checkArgument(
```
Comment (jerryshao) [inline catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveCatalogOperations.java:882]: I found that some of the changes here and below change the semantics (change to the different exception), I would  suggest we don't do this change and keep as it was.

File: common/src/main/java/com/datastrato/gravitino/dto/rel/ColumnDTO.java [hunk 2]
```diff
@@ -228,17 +229,13 @@ public ColumnDTO build() {
...
-      throw new IllegalArgumentException("Column data type cannot be null.");
-    }
-    if (!nullable()
-        && defaultValue() instanceof LiteralDTO
-        && ((LiteralDTO) defaultValue()).dataType().equals(Types.NullType.get())) {
-      throw new IllegalArgumentException(
-          "Column cannot be non-nullable with a null default value: " + name());
-    }
+    Preconditions.checkArgument(
+        !Strings.isNullOrEmpty(name()), "Column name cannot be null or empty");
```
Comment (yuqi1129) [inline common/src/main/java/com/datastrato/gravitino/dto/rel/ColumnDTO.java:233]: `StringUtils.isNotEmpty()` may be better. This makes it difficult for users to comprehend `!`.

**Comment by yuqi1129** [comment]
Other changes are acceptable to me.

**Review by yuqi1129 — COMMENTED** [review]
LGTM

**Review by yuqi1129 — APPROVED** [review]

**Comment by yuqi1129** [comment]
@jerryshao 
Do you have any further comments?

**Comment by qqqttt123** [comment]
Could we have a checker tool to ensure that we have fixed all the issues?

**Comment by SteNicholas** [comment]
@qqqttt123, this could create new ticket for this. No need to add the checker tool in this pull request.

**Comment by jerryshao** [comment]
I don't think it is possible to add a check tool for this issue, this is more like a style thing that can be improved but also acceptable for the `if..else` style.

**Comment by qqqttt123** [comment]
LGTM.

**Review by qqqttt123 — APPROVED** [review]

**Comment by yuqi1129** [comment]
@jerryshao @qqqttt123 Thanks for your kind review.
SteNicholas, thank you for your contributions.

**Comment by jerryshao** [comment]
I think the current fix overly changes all the if...else logic to use `Preconditions.checkArgument(xxx)`, which makes the readability of the current change super hard.

The purpose of `Preconditions.checkArgument(xxx)` is to check the validity of the argument, not the validity of the logic, overly using this will make people hard to understand the code.

**Comment by jerryshao** [comment]
> I think the current fix overly changes all the if...else logic to use `Preconditions.checkArgument(xxx)`, which makes the readability of the current change super hard.
> 
> The purpose of `Preconditions.checkArgument(xxx)` is to check the validity of the input arguments, not the validity of the logic, overly using this will make people hard to understand the code, also make the logic quite broken.
> 
> So I would suggest reverting the current commit and submitting a new PR to fix this issue prudently. What's your opinion? @SteNicholas

@SteNicholas I'm going to revert this PR. Would you please resubmit a new PR as I mentioned above when you have time? Thanks a lot.

**Comment by SteNicholas** [comment]
@jerryshao, I will resubmit a PR to follow above comment.

**Pull request merged**